### DISCUSSION TITLE
New permission for actions token: `labels: write`

### DISCUSSION THREAD

**Discussion by jozefizso** [community/community#13565] [Actions]
I have written GitHub Action which modifies **labels** in the repository. The auth token set by the `issues: write` permission does not allow creating/updating/deleting of labels in the repository, despite the API for labels being categorized under issues.

As a workaround, the personal access token write `repo` permissions must be created and used as authentication. **Such PAT has too broad permissions.**

To use the least privilege principle a new `labels: write` permissions should be supported by GitHub Actions token.

See https://github.community/t/create-and-update-labels-in-repository-from-action/240818

**Comment by ethomson** [comment]
Thanks for the feedback @jozefizso - I'll give this feedback to the team that owns access tokens and permissions so that they can track this work item. Once we have tokens that can be scoped to a label, we can take a look at adding that scope to the `GITHUB_TOKEN`.

Thanks again!

**Reply by davorpa** [reply]
Hey @ethomson, any feedback about this feature?

**Reply by cpanato** [reply]
any news? thanks!

**Comment by daltonv** [comment]
To add onto this suggestion I would also change the PAT option so that there is a `runners: write` scope for BOTH repos and organizations. 

I often use this ansible role (https://github.com/MonolithProjects/ansible-github_actions_runner) to add new runners to a github organization and I have to use a PAT with the **very** powerful `adming:org` scope. I would feel much better about automating these deployments if the PAT had a more limited scope.

**Comment by stevenengland** [comment]
> I have written GitHub Action which modifies **labels** in the repository. The auth token set by the `issues: write` permission does not allow creating/updating/deleting of labels in the repository, despite the API for labels being categorized under issues.
> 
> As a workaround, the personal access token write `repo` permissions must be created and used as authentication. **Such PAT has too broad permissions.**
> 
> To use the least privilege principle a new `labels: write` permissions should be supported by GitHub Actions token.
> 
> See https://github.community/t/create-and-update-labels-in-repository-from-action/240818

Seems that at least **issues: write** now contains the permission to edit labels. I was able to create and update labels with the GITHUB_TOKEN with this permission granted.

**Comment by Jonty** [comment]
For anyone else searching for this, it seems to have been broken: https://github.com/orgs/community/discussions/149877
